File: client/src/types/prompt.ts
/**
 * Prompt Management Types
 *
 * TypeScript types for the prompt library feature.
 */

// Valid categories for prompts
export type PromptCategory =
  | 'customer'
  | 'inventory'
  | 'analytics'
  | 'reporting'
  | 'general';

// Prompt entity
export interface Prompt {
  id: number;
  title: string;
  content: string;
  category: PromptCategory;
  description: string | null;
  is_favorite: boolean;
  usage_count: number;
  last_used_at: string | null;
  created_at: string;
  updated_at: string;
}

// Request to create a new prompt
export interface PromptCreateRequest {
  title: string;
  content: string;
  category: PromptCategory;
  description?: string;
  is_favorite?: boolean;
}

// Request to update a prompt
export interface PromptUpdateRequest {
  title?: string;
  content?: string;
  category?: PromptCategory;
  description?: string;
  is_favorite?: boolean;
}

// Response containing list of prompts
export interface PromptListResponse {
  prompts: Prompt[];
  total: number;
}

// Response after incrementing usage
export interface PromptUsageResponse {
  id: number;
  usage_count: number;
  last_used_at: string;
}

// Category display configuration
export interface CategoryConfig {
  value: PromptCategory;
  label: string;
  icon: string;
}

// Category configurations for UI
export const PROMPT_CATEGORIES: CategoryConfig[] = [
  { value: 'customer', label: 'Customer', icon: '👥' },
  { value: 'inventory', label: 'Inventory', icon: '📦' },
  { value: 'analytics', label: 'Analytics', icon: '📊' },
  { value: 'reporting', label: 'Reporting', icon: '📈' },
  { value: 'general', label: 'General', icon: '💬' },
];

// Category label lookup
export const CATEGORY_LABELS: Record<PromptCategory, string> = {
  customer: 'Customer',
  inventory: 'Inventory',
  analytics: 'Analytics',
  reporting: 'Reporting',
  general: 'General',
};

// Category icon lookup
export const CATEGORY_ICONS: Record<PromptCategory, string> = {
  customer: '👥',
  inventory: '📦',
  analytics: '📊',
  reporting: '📈',
  general: '💬',
};

File: server/schemas/prompt.py
"""
Pydantic schemas for Prompt API.

Defines request and response models for prompt management endpoints.
"""

from datetime import datetime
from typing import Literal, Optional
from pydantic import BaseModel, Field


# Valid categories for prompts
PromptCategory = Literal["customer", "inventory", "analytics", "reporting", "general"]


# ==================== Request Models ====================


class PromptCreateRequest(BaseModel):
    """Request to create a new prompt."""

    title: str = Field(..., min_length=1, max_length=255, description="Unique title for the prompt")
    content: str = Field(..., min_length=1, description="The prompt text content")
    category: PromptCategory = Field(..., description="Category for organizing prompts")
    description: Optional[str] = Field(None, description="Optional description of the prompt")
    is_favorite: bool = Field(default=False, description="Whether to mark as favorite")


class PromptUpdateRequest(BaseModel):
    """Request to update a prompt. All fields are optional."""

    title: Optional[str] = Field(None, min_length=1, max_length=255)
    content: Optional[str] = Field(None, min_length=1)
    category: Optional[PromptCategory] = None
    description: Optional[str] = None
    is_favorite: Optional[bool] = None


# ==================== Response Models ====================


class PromptResponse(BaseModel):
    """A prompt template."""

    id: int
    title: str
    content: str
    category: str
    description: Optional[str]
    is_favorite: bool
    usage_count: int
    last_used_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PromptListResponse(BaseModel):
    """List of prompts."""

    prompts: list[PromptResponse]
    total: int


class PromptUsageResponse(BaseModel):
    """Response after incrementing usage."""

    id: int
    usage_count: int
    last_used_at: datetime
